lib/omnibus/packagers/fastmsi.rb

```diff
@@ -0,0 +1,654 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'pathname'
+
+module Omnibus
+  class Packager::FastMSI < Packager::Base
+    DEFAULT_TIMESTAMP_SERVERS = ['http://timestamp.digicert.com',
+                                 'http://timestamp.verisign.com/scripts/timestamp.dll']
+    id :fastmsi
+
+    setup do
+      # Render the localization
+      write_localization_file
+
+      # Render the msi parameters
+      write_parameters_file
+
+      # Render the source file
+      write_source_file
+
+      # Render the custom action file
+      write_custom_action_file
+
+      # Optionally, render the bundle file
+      write_bundle_file if bundle_msi
+
+      # Copy all the staging assets from vendored Omnibus into the resources
+      # directory.
+      create_directory("#{resources_dir}/assets")
+      FileSyncer.glob("#{Omnibus.source_root}/resources/#{id}/assets/*").each do |file|
+        copy_file(file, "#{resources_dir}/assets/#{File.basename(file)}")
+      end
+
+      # Copy all assets in the user's project directory - this may overwrite
+      # files copied in the previous step, but that's okay :)
+      FileSyncer.glob("#{resources_path}/assets/*").each do |file|
+        copy_file(file, "#{resources_dir}/assets/#{File.basename(file)}")
+      end
+
+      # Copy externally built custom action files into the staging directory
+      FileSyncer.glob("#{resources_path}/ext/*").each do |file|
+        copy_file(file, "#{staging_dir}")
+      end
+    end
+
+    build do
+      # Zip up the contents of the install directory
+      zip_command = <<-EOH.split.join(' ').squeeze(' ').strip
+        7z a -r
+          #{windows_safe_path(staging_dir)}\\#{project.name}.zip
+          #{windows_safe_path(project.install_dir)}\\*
+      EOH
+      shellout!(zip_command, returns: [0])
+
+      # Harvest the zip file with heat.exe
+      Dir.chdir(staging_dir) do
+        heat_command = <<-EOH.split.join(' ').squeeze(' ').strip
+          heat.exe file "#{project.name}.zip"
+            -cg ProjectDir
+            -dr INSTALLLOCATION
+            -nologo -sfrag -srd -sreg -gg
+            -out "project-files.wxs"
+        EOH
+        shellout!(heat_command, returns: [0])
+
+        # Compile with candle.exe
+        log.debug(log_key) { "wix_candle_flags: #{wix_candle_flags}" }
+
+        candle_command = <<-EOH.split.join(' ').squeeze(' ').strip
+          candle.exe
+            -nologo
+            #{wix_candle_flags}
+            #{wix_extension_switches(wix_candle_extensions)}
+            -dProjectSourceDir="#{windows_safe_path(project.install_dir)}" "project-files.wxs"
```

Comment:
No ProjectSourceDir anymore.  If we have a zip, we shouldn't be referring to source files.